spec/src/main/asciidoc/configuration.adoc

```diff
@@ -0,0 +1,307 @@
+//
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+[[sec:configuration-properties]]
+== Configuration
+
+OpenTelemetry MUST be configured by MicroProfile Config following the semantics of configuration properties detailed in https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}/sdk-extensions/autoconfigure[OpenTelemetry SDK Autoconfigure {otel-java-version}].
+Following properties MUST be supported:
+
+=== Configuration Properties
```

Comment:
Supporting all these properties from Autoconfigure might be a bit of a stretch... Some of them can be optionally implemented. Trace and Metrics specific properties are just overrides to the general case and in most cases they should not be mandatory. 
This will reduce the implementation burden.

Reply:
As long we are making this a 2.0 release we could potentially make support for some of these optional, but these needs wider consensus, and a separate issue. Should we just drop per-signal endpoints, or maybe certificate authentication as well?

Reply:
Per signal endpoints maybe but not the certificate ones.

Reply:
Sorry, the other way around. We should require the certificate ones but not the per-signal endpoints. Basically all redundant properties should be optional. By redundant I mean the ones having a parent property that can be assumed.

Reply:
Per signal OTLP exporter configuration parameters is now mentioned as notable optional ones instead.

Reply:
@pdudits will you update this PR to remove the properties but only mention which properties are not mandatory? What is the consensus here? Do we want to explicitly list them and maintain them or just add a link to the OTel site at a particular version we are align with? I prefer the latter for maintenance purpose.

Reply:
Comparing to https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure, we require support for about 25 properties out of around 80 mentioned on that page.

Therefore it is much clearer to enumerate the 25 required properties for both users and implementors. Especially since we speak about MP Config properties rather than just system properties, and the formatting on github page makes it quite hard to read the description.

I'm fine with dropping the optional ones completely.

Reply:
I don't think that change should be part of this PR, @Emily-Jiang. Configuration properties were already defined in 1.1 and are only lightly extended for those required for metrics.

Reworking the means of configuration deserves its own PR.

Reply:
Maybe you don't do any changes in the config section. I think we are better off not to repeat the same info as the OTel config section because we support most configs. We just need to point to the section and highlight the configs we don't support.